assets/app/scripts/directives/deleteButton.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .directive("deleteButton", function ($modal, $location, $filter, hashSizeFilter, DataService, AlertMessageService, Logger) {
+    return {
+      restrict: "E",
+      scope: {
+        resourceType: "@",
+        resourceName: "@",
+        projectName: "@",
+        alerts: "=alerts",
+        displayName: "@"
+      },
+      templateUrl: "views/directives/delete-button.html",
+      link: function(scope, element, attrs) {
+        // make resource types available in the modal
+        scope.resourceType = attrs.resourceType;
+        scope.resourceName = attrs.resourceName;
+        scope.projectName = attrs.projectName;
+        scope.displayName = attrs.displayName;
+
+        if (attrs.resourceType === 'project') {
+          scope.isProject = true;
+          scope.biggerButton = true;
+        }
+
+        // make the resource types show up nicely in the modal
+        scope.nameFormatMap = {
```

Comment:
create a filter called humanizeResourceType or humanizeKind that takes the kind and returns the humanized name so that we can reuse this later